web-client/src/presenter/computeds/formattedCaseDetail.js

```diff
@@ -7,10 +7,63 @@ export const formatDocument = document => {
   result.createdAtFormatted = moment(result.createdAt).format('L');
   result.showValidationInput = !result.reviewDate;
   result.isStatusServed = result.status === 'served';
+  result.isPetition = result.documentType === 'Petition';
   return result;
 };
 
-const formatCase = caseDetail => {
+export const formatYearAmounts = (caseDetail, caseDetailErrors = {}) => {
+  caseDetail.canAddYearAmount =
+    (caseDetail.yearAmounts || []).filter(yearAmount => {
+      return !yearAmount.year;
+    }).length !== 1;
+
+  if (!caseDetail.yearAmounts || caseDetail.yearAmounts.length === 0) {
```

Comment:
Can this be broken into smaller components? It's getting a bit complex.

Reply:
I refactored this some for you

Reply:
Lovely, thank you.